_query-dsl/specialized/percolate.md

```diff
@@ -0,0 +1,704 @@
+---
+layout: default
+title: Percolate
+parent: Specialized queries
+nav_order: 55
+---
+
+# Percolate
+
+Use the `percolate` query to find stored queries that match a given document. This is the opposite of a regular search. Instead of finding documents that match a query, you find queries that match a document. Percolate queries are often used for alerting, notifications, and reverse search use cases.
+
+- You can percolate a document provided inline or fetch an existing document from an index.
+- The document and the stored queries must use the same field names and types.
+- You can combine percolation with filtering and scoring to build complex matching systems.
+- Percolate queries are considered expensive and will only run if the cluster setting `search.allow_expensive_queries` is set to `true` (default). If this setting is `false`, percolate queries will be rejected.
+
+Percolate queries are useful in a variety of real-time matching scenarios. Some common use cases include:
+
+- **E-commerce notifications**: Customers can register interest in products, for example: “Notify me when new Apple laptops are in stock”. When new product documents are indexed, the system finds all users with matching saved queries and sends alerts.
+- **Job alerts**: Job seekers save queries based on preferred job titles or locations, and new job postings are matched against these to trigger alerts.
+- **Security and alerting systems**: Percolate incoming log or event data against saved rules or anomaly patterns.
+- **News filtering**: Match incoming articles against saved topic profiles to categorize or deliver relevant content.
+
+## How percolation works
+
+1. Saved queries are stored in a special [`percolator` field type]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/percolator/).
+2. Documents are compared against all saved queries.
+3. Matching queries are returned with their `_id`.
+4. If highlighting is enabled, matched text snippets are also returned.
+5. If multiple documents are sent, `_percolator_document_slot` displays which document matched.
+
```

Comment:
Let's add a heading level 2 "Example", an intro sentence for the example, and make the steps heading level 3.